epictrack-api/src/api/resources/work_status.py

```diff
@@ -35,5 +36,56 @@ class WorkStatus(Resource):
     @auth.require
     def get(work_id):
         """Return all active works."""
-        works = WorkStatusService.find_all_works_tatus(work_id)
+        works = WorkStatusService.find_all_work_status(work_id)
         return jsonify(res.WorkStatusResponseSchema(many=True).dump(works)), HTTPStatus.OK
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def post(work_id):
+        """Create new work status"""
+        request_dict = req.WorkStatusParameterSchema().load(API.payload)
+        work_status = WorkStatusService.create_work_status(work_id, request_dict)
+        return res.WorkStatusResponseSchema().dump(work_status), HTTPStatus.CREATED
+
+
+@cors_preflight("GET, PUT")
+@API.route("/<int:status_id>", methods=["GET", "PUT", "OPTIONS"])
+class Status(Resource):
+    """Endpoint resource to manage a work status."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def put(work_id, status_id):
+        """Update work status"""
+        request_dict = req.WorkStatusParameterSchema().load(API.payload)
+        existing_work_status = WorkStatusService.find_work_status_by_id(work_id, status_id)
+        if existing_work_status is None:
+            return {"message": "Work status not found"}, HTTPStatus.NOT_FOUND
+
+        updated_work_status = WorkStatusService.update_work_status(existing_work_status, request_dict)
+
+        return res.WorkStatusResponseSchema().dump(updated_work_status), HTTPStatus.OK
+
+
+@cors_preflight("PATCH")
+@API.route("/<int:status_id>/approve", methods=["PATCH", "OPTIONS"])
+class ApproveStatus(Resource):
+    """Endpoint resource to manage approving of work status."""
+
+    @staticmethod
+    @cors.crossdomain(origin="*")
+    @auth.require
+    @profiletime
+    def patch(work_id, status_id):
```

Comment:
We used to Create a path parameter schema to validate the path parameters. Could you please check if that is required at all

Reply:
hm..we can discuss this..